userspace/libscap/engine/gvisor/gvisor.h

```diff
@@ -73,5 +78,51 @@ typedef struct parse_result parse_result;
 */
 parse_result parse_gvisor_proto(scap_const_sized_buffer gvisor_buf, scap_sized_buffer scap_buf);
 
+uint64_t get_vxid(uint64_t vxid);
+
 } // namespace parsers
+
+// contains entries to store per-sandbox data and buffers to use to write events in
+class sandbox_entry {
+public:
+	sandbox_entry();
+    ~sandbox_entry();
+
+    int32_t expand_buffer(size_t size);
+
+	scap_sized_buffer m_buf;
+};
```

Comment:
I like the idea by @gnosek about adding this. Just out of curiosity: can't we use a `vector<char>` instead? To allocate the initial memory we could just use `reserve`, to reallocate memory `resize`, and to access the buffer and size `data` and `size`. Also, there will be no need to implement any destructor. Wdyt @LucaGuerra ?